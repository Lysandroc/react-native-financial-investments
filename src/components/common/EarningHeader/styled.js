import styled from 'styled-components/native';
import Text from '../Text';

export const WrapperView = styled.View`
	flex-basis: 20%;
	flex-direction: row;
	flex-wrap: wrap;
	align-content: flex-end;
`;

export const Earning = styled(Text)`
	flex: 1 100%;
	color: #505d65;
	font-size: 42px;
`;

export const Percentage = styled(Text)`
	flex: 1 50%;
	color: green;
`;

export const MoneyInvested = styled(Text)`
	flex: 1 50%;
	color: #505d65;
`;
