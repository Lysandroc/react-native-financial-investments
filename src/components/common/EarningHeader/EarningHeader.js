import React from 'react';
import { WrapperView, Earning, Percentage, MoneyInvested } from './styled';

import { currency } from '../../../utils/masks';

const EarningHeader = ({ amountInvested = 0 }) => {
	return (
		<WrapperView>
			<Earning dark center accessibilityLabel="earning-value">
				R$ {currency.mask(amountInvested)}
			</Earning>
			<Percentage bold large center accessibilityLabel="percentage-value">
				+150%
			</Percentage>
			<MoneyInvested large accessibilityLabel="earning-invested">
				R$ {currency.mask(amountInvested / 2.5)}
			</MoneyInvested>
		</WrapperView>
	);
};

export default EarningHeader;
