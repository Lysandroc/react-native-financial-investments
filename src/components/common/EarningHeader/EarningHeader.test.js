import React from 'react';
import { render, wait, cleanup } from '@testing-library/react-native';
import EarningHeader from '../EarningHeader';

describe('should test <HeaderImage/>', () => {
	let component;
	let props = { amountInvested: 1000.0 };

	beforeEach(() => {
		jest.clearAllMocks();
		component = render(<EarningHeader {...props} />);
	});

	afterEach(cleanup);

	it('snapshot should be valid', () => {
		expect(component.baseElement).toMatchSnapshot();
	});

	it('should return earning value correctly', () => {
		const earningValue = component.queryByLabelText('earning-value');
		expect(earningValue).toHaveTextContent('R$ 1.000,00');
	});

	it('should return percentage correctly', () => {
		const percentageValue = component.queryByLabelText('percentage-value');
		expect(percentageValue).toHaveTextContent('+150%');
	});

	it('should return earning invested correctly', () => {
		const earningInvested = component.queryByLabelText('earning-invested');
		expect(earningInvested).toHaveTextContent('R$ 400,00');
	});
});
