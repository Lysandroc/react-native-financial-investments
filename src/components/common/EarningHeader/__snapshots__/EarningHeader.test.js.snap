// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test <HeaderImage/> snapshot should be valid 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignContent": "flex-end",
            "flexBasis": "20%",
            "flexDirection": "row",
            "flexWrap": "wrap",
          },
        ]
      }
    >
      <Text
        accessibilityLabel="earning-value"
        center={true}
        dark={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "color": "#505d65",
              "flexBasis": "100%",
              "flexGrow": 1,
              "flexShrink": 1,
              "fontFamily": "Poppins-Regular",
              "fontSize": 42,
              "fontWeight": "normal",
              "justifyContent": "center",
              "textAlign": "center",
            },
          ]
        }
      >
        R$ 
        1.000,00
      </Text>
      <Text
        accessibilityLabel="percentage-value"
        bold={true}
        center={true}
        large={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "color": "green",
              "flexBasis": "50%",
              "flexGrow": 1,
              "flexShrink": 1,
              "fontFamily": "Poppins-Regular",
              "fontSize": 16,
              "fontWeight": "bold",
              "justifyContent": "center",
              "textAlign": "center",
            },
          ]
        }
      >
        +150%
      </Text>
      <Text
        accessibilityLabel="earning-invested"
        large={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "color": "#505d65",
              "flexBasis": "50%",
              "flexGrow": 1,
              "flexShrink": 1,
              "fontFamily": "Poppins-Regular",
              "fontSize": 16,
              "fontWeight": "normal",
              "justifyContent": "center",
              "textAlign": "left",
            },
          ]
        }
      >
        R$ 
        400,00
      </Text>
    </View>
  </View>
</View>
`;
